src/components/ArticleList.jsx

```diff
@@ -0,0 +1,45 @@
+import axios from "axios";
+import { useEffect, useState } from "react";
+import SingleArticle from "./SingleArticle";
+import { Link } from "react-router-dom";
+
+const ArticleList = () => {
+    const [articles, setArticles] = useState([])
+    const [isLoading, setIsLoading] = useState(true)
+
+    const getArticles = async () => {
```

Comment:
API requests should be handled separately in a utils file. I'd extract this out 👍 You can also set up a baseURL using axios.create()